gubernator/main.py

```diff
@@ -191,30 +191,117 @@ def build_details(build_dir):
 
 @memcache_memoize('log-file://', expires=60*60*4)
 def find_log((build_dir, junit, log_file)):
-    tmps = [f.filename for f in gcs_ls('%s/artifacts' % build_dir)
-            if '/tmp-node' in f.filename]
-    for folder in tmps:
-        filenames = [f.filename for f in gcs_ls(folder)]
-        if folder + junit in filenames:
-            path = folder + log_file
-            if path in filenames:
-                return path
+    '''
+    Given a junit file, looks in build_dir for log_file in a folder that
+    also includes the junit file.
+    If not given a junit file, returns list of files named log_file in
+    all folders in build_dir.
+    '''
+    if junit:
+        tmps = [f.filename for f in gcs_ls('%s/artifacts' % build_dir)
+                if '/tmp-node' in f.filename]
+        for folder in tmps:
+            filenames = [f.filename for f in gcs_ls(folder)]
+            if folder + junit in filenames:
+                path = folder + log_file
+                if path in filenames:
+                    return path
+    else:
```

Comment:
split this into two functions, depending on whether there's a junit file or not.